src/react/hooks/useBackgroundQuery.ts

```diff
@@ -202,13 +206,16 @@ export function useBackgroundQuery<
     client.watchQuery(watchQueryOptions as WatchQueryOptions<any, any>)
   );
 
-  const [promiseCache, setPromiseCache] = React.useState(
-    () => new Map([[queryRef.key, queryRef.promise]])
-  );
-
+  const [wrapped, setWrappedQueryRef] = React.useState({
+    current: wrapQueryRef(queryRef, queryRef.promise),
+  });
```

Comment:
I'm curious about the decision to create this as an object with a `current` property. It doesn't look like you do any mutable update to the `current` property and exclusively rely on `setWrappedQueryRef`.

Since `wrapQueryRef` will return a new object each time, that should be sufficient enough to ensure React won't bail on the render when comparing old and new values. Could you consider simplifying this to just set the wrapped query ref directly?

```suggestion
  const [wrappedQueryRef, setWrappedQueryRef] = React.useState(() => 
    wrapQueryRef(queryRef, queryRef.promise)
  );
```